server.go

```diff
@@ -203,47 +203,77 @@ func newServer(listenAddrs []string, chanDB *channeldb.DB, cc *chainControl,
 		selfAddrs = append(selfAddrs, lnAddr)
 	}
 
-	chanGraph := chanDB.ChannelGraph()
-
 	// TODO(roasbeef): make alias configurable
 	alias, err := lnwire.NewNodeAlias(hex.EncodeToString(serializedPubKey[:10]))
 	if err != nil {
 		return nil, err
 	}
-	selfNode := &channeldb.LightningNode{
```

Comment:
I kind of like having this up here. It makes more sense to me that a NodeAnnouncement would be derived from a LightningNode than constructing it directly. In fact, `LightningNode` could have a method that constructs a `NodeAnnouncement`, which you could use to get both `nodeAnn` and `oldNodeAnn`.